darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,691 @@
+                series_subset += [
+                    (s.univariate_component(c), transformation, builtins)
+                    for s in series
+                    for c in range(s.width)
+                ]
+
+            elif "series_id" in transformation and "comp_id" not in transformation:
+                # apply the transformation to a specific series and all its components
+                raise_if_not(
+                    len(series) - 1 >= max(transformation["series_id"]),
+                    f"`window_transformation` at index {idx} has a 'series_id' that is greater than "
+                    f"the number of series. ",
+                )
+                series_subset += [
+                    (series[s_idx].univariate_component(c), transformation, builtins)
+                    for s_idx in transformation["series_id"]
+                    for c in range(series[s_idx].width)
+                ]
+
+            elif "series_id" not in transformation and "comp_id" in transformation:
+                # apply the transformation to all series on a specific component only in each series
+                # test that component exists in each relevant series
+                raise_if_not(
+                    all(
+                        c_idx <= series[s_idx].width - 1
+                        for s_idx in range(len(series))
+                        for c_idx in transformation["comp_id"]
+                    ),
+                    "Some components are not available in the provided series.",
+                )
+
+                series_subset += [
+                    (s.univariate_component(c), transformation, builtins)
+                    for (s, c) in itertools.product(series, transformation["comp_id"])
+                ]
+
+            else:
+                # apply the transformation to a specific component in a specific series
+                # if a different component is provided for each selected series
+                if all(isinstance(x, list) for x in transformation["comp_id"]) and len(
+                    transformation["comp_id"]
+                ) == len(transformation["series_id"]):
+                    # test that components exist in the corresponding series
+                    raise_if_not(
+                        all(
+                            max(c_idxvec) <= series[s_idx].width - 1
+                            for (s_idx, c_idxvec) in zip(
+                                transformation["series_id"], transformation["comp_id"]
+                            )
+                        ),
+                        "Some components are not available in the provided series.",
+                    )
+                    series_subset += [
+                        (
+                            series[s_idx].univariate_component(c_idx),
+                            transformation,
+                            builtins,
+                        )
+                        for (s_idx, c_idx) in itertools.chain(
```

Comment:
This is a little hard to read 😅 Some intermediate variables could be welcome.

Reply:
simplified with the new version, should be clearer